rbd.c

```diff
@@ -1484,6 +1563,28 @@ static int tcmu_rbd_reconfig(struct tcmu_device *dev,
 	}
 }
 
+static void tcmu_rbd_destroy(void)
+{
+	char **entry;
+	int i = 0;
+
+	tcmu_info("destroying the rbd handler\n");
+	pthread_mutex_lock(&blacklist_caches_lock);
+	if (darray_empty(blacklist_caches))
+		goto unlock;
+
+	/* Try to remove all the stale blacklist entities */
+	darray_foreach(entry, blacklist_caches) {
+		free(*entry);
+		i++;
+	}
+
+	darray_free(blacklist_caches);
```

Comment:
Perhaps you can have an `init` handler to mimic the `destroy` handler to avoid the asymmetric initialization of this array vs its destruction?

Reply:
Sound good. I will do it.